internal/expr/parser.go

```diff
@@ -502,35 +574,104 @@ func (p *Parser) parseColumn() (columnAccessor, bool, error) {
 	return columnAccessor{}, false, nil
 }
 
-func (p *Parser) parseTargetType() (valueAccessor, bool, error) {
+func (p *Parser) parseTargetType() (memberAcessor, bool, error) {
+	startLine := p.lineNum
+	startCol := p.colNum()
+
 	if p.skipByte('&') {
+		// Using a slice as an output is an error, we add the case here to
+		// improve the error message.
+		if st, ok, err := p.parseSliceAccessor(); err != nil {
+			return memberAcessor{}, false, err
+		} else if ok {
+			return memberAcessor{}, false, errorAt(fmt.Errorf("cannot use slice syntax %q in output expression", st), startLine, startCol, p.input)
+		}
 		return p.parseTypeName()
 	}
 
-	return valueAccessor{}, false, nil
+	return memberAcessor{}, false, nil
+}
+
+// parseUNumber parses a non-negative number composed of one or more digits.
```

Comment:
Maybe worth pointing out why this value is a pointer rather than a straight integer here. Or changing it to return a plain integer and doing the pointer bits in the call site.